CHANGELOG.md

```diff
@@ -1,5 +1,30 @@
 # Changelog
 
+## `v10.2.0`
+
+### Miscellaneous small additions and changes
+
+- **[ci]** improved caching [#2197](https://github.com/docker-mailserver/docker-mailserver/pull/2197)
+- **[ci]** refactored spam tests and introduced common container setup template [#2198](https://github.com/docker-mailserver/docker-mailserver/pull/2198)
+- **[fix]** update Fail2Ban wrapper to propagate errors to user [#2170](https://github.com/docker-mailserver/docker-mailserver/pull/2170)
+- **[fix]** Dockerfile `sed`'s are now checked [#2158](https://github.com/docker-mailserver/docker-mailserver/pull/2158)
+- **[general]** Updated default value of `ONE_DIR` to `1` [#2148](https://github.com/docker-mailserver/docker-mailserver/pull/2148)
+- **[docs]** updated Kubernetes documentation [#2111](https://github.com/docker-mailserver/docker-mailserver/pull/2111)
+- **[docs]** introduced dedicated Podman documentation [#2179](https://github.com/docker-mailserver/docker-mailserver/pull/2179)
+- **[docs]** miscellaneous documentation improvements
+- **[misc]** introduced GitHub issue forms for  issue templates [#2160](https://github.com/docker-mailserver/docker-mailserver/pull/2160)
+- **[misc]** removed old `mkcert.sh` [#2196](https://github.com/docker-mailserver/docker-mailserver/pull/2196)
+
+### Bigger scripts-related improvements
+
+- **[scripts]** update `setup.sh` to now use a running container first if one exists [#2134](https://github.com/docker-mailserver/docker-mailserver/pull/2134)
+- **[scripts]** included `setup.sh` functionality inside the container to be version independent again [#2174](https://github.com/docker-mailserver/docker-mailserver/pull/2174)
+- **[scripts]** `HOSTNAME` and `DOMAINNAME` setup improved [#2175](https://github.com/docker-mailserver/docker-mailserver/pull/2175)
+- **[scripts]** `delmailuser` can now delete mailboxed without TLD [#2172](https://github.com/docker-mailserver/docker-mailserver/pull/2172)
+- **[scripts]** refactored `_setup_ssl` in `setup-stack.sh` [#2196](https://github.com/docker-mailserver/docker-mailserver/pull/2196)
```

Comment:
Suggestions if you want to add any to release notes:

---

> An invalid `SSL_TYPE` or a valid value with invalid configuration will now panic, exiting the container and emitting a fatal error to the logs.

> An unconfigured/empty `SSL_TYPE` ENV now correctly disables SSL support for Dovecot and general Postfix configurations. A reminder that this is unsupported officially, and is only intended for tests and troubleshooting. Use only [a valid `SSL_TYPE`](https://docker-mailserver.github.io/docker-mailserver/v10.2/config/environment/#ssl_type) (_`letsencrypt` and `manual` are recommended_) for production deployments.

> For anyone relying on internal location of certificates (_internal copy of mounted files at startup_), the Postfix and Dovecot location of `/etc/postfix/ssl` has changed to `/etc/dms/tls`. This may affect any third-party `user-patches.sh` scripts that depended on this path to update certs.

> `TLS_LEVEL=intermediate` will modify the system (container) `openssl.cnf` to set a minimum protocol of TLS 1.0 (_from 1.2_) and ciphersuite support to `DEFAULT@SECLEVEL=1` (_from `2`_). This change is required for Dovecot in upcoming Debian Bullseye upgrade, to be compatible with the `TLS_LEVEL=intermediate` cipher suite profile. It may affect other software within the container that relies on this openssl config, should you extend the Docker image.

<details>
<summary>Markdown snippets</summary>

```markdown
- An invalid `SSL_TYPE` or a valid value with invalid configuration will now panic, exiting the container and emitting a fatal error to the logs.

- An unconfigured/empty `SSL_TYPE` ENV now correctly disables SSL support for Dovecot and general Postfix configurations. A reminder that this is unsupported officially, and is only intended for tests and troubleshooting. Use only [a valid `SSL_TYPE`](https://docker-mailserver.github.io/docker-mailserver/v10.2/config/environment/#ssl_type) (_`letsencrypt` and `manual` are recommended_) for production deployments.

- For anyone relying on internal location of certificates (_internal copy of mounted files at startup_), the Postfix and Dovecot location of `/etc/postfix/ssl` has changed to `/etc/dms/tls`. This may affect any third-party `user-patches.sh` scripts that depended on this path to update certs.

- `TLS_LEVEL=intermediate` will modify the system (container) `openssl.cnf` to set a minimum protocol of TLS 1.0 (_from 1.2_) and ciphersuite support to `DEFAULT@SECLEVEL=1` (_from `2`_). This change is required for Dovecot in upcoming Debian Bullseye upgrade, to be compatible with the `TLS_LEVEL=intermediate` cipher suite profile. It may affect other software within the container that relies on this openssl config, should you extend the Docker image.
```

</details>